src/main/java/com/alex/eshop/keycloak/KeycloakService.java

```diff
@@ -25,4 +41,66 @@ public AccessTokenResponse getToken() {
         AuthzClient authzClient = AuthzClient.create(authzConf);
         return authzClient.obtainAccessToken();
     }
+
+    public RealmResource realmResource() {
+        return keycloakClientFactory.getInstance()
+                .realm(applicationProperties.getKeycloak().getRealm());
+    }
+
+    public UsersResource usersResource() {
+        return keycloakClientFactory.getInstance()
+                .realm(applicationProperties.getKeycloak().getRealm())
+                .users();
+    }
+
+    public List<RoleRepresentation> getRoleRepresentation() {
+        List<RoleRepresentation> realmRoles = new ArrayList<>();
+
+        realmRoles.add(realmResource().roles().get(Role.USER).toRepresentation());
+
+        return realmRoles;
+    }
+
+    public List<String> getUserRole() {
+        return getRoleRepresentation().get(0).getName().lines().toList();
+    }
+
+    public UserRepresentation getUserRepresentation(String userUuid) {
+        return usersResource()
+                .get(userUuid)
+                .toRepresentation();
+    }
+
+    public UserRepresentation createUser(UserRegisterDTO userRegisterDTO) {
+        if (!Objects.equals(userRegisterDTO.getPassword(), userRegisterDTO.getConfirmPassword())) {
+            throw new InvalidDataException("Invalid password");
+        }
+
+        CredentialRepresentation credentialRepresentation = createPasswordCredentials(userRegisterDTO.getPassword());
+
+        UserRepresentation userRepresentation = new UserRepresentation();
+
+        userRepresentation.setUsername(userRegisterDTO.getUserName());
+        userRepresentation.setEmail(userRegisterDTO.getEmail());
+        userRepresentation.setFirstName(userRegisterDTO.getFirstName());
+        userRepresentation.setLastName(userRegisterDTO.getLastName());
+        userRepresentation.setCredentials(Collections.singletonList(credentialRepresentation));
+        userRepresentation.setEnabled(true);
+
+        if (usersResource().create(userRepresentation).getStatus() != 201) {
```

Comment:
this is poor status handling. I found at least 3 different return statuses: CREATED, BAD_REQUEST, CONFLICT.
You can use switch-case to handle different statuses to provide detailed info what happened. 

```
// class field
private static final Logger logger = LoggerFactory.getLogger(KeycloakService.class);

// instead lines 90-92
Response response = usersResource().create(userRepresentation);
HttpStatus httpStatus = HttpStatus.valueOf(response.getStatus());
switch (httpStatus) {
    case CREATED -> logger.info("User {} successfully created", userRegisterDTO);
    case BAD_REQUEST -> throw new InvalidDataException("Cannot create user " + userRegisterDTO);
    case CONFLICT -> throw new ConflictException("User " + userRegisterDTO + " already exists");
    default -> throw new ResponseStatusException(httpStatus, "Problems occurred while creating user " + userRegisterDTO);
}
```

also you can add `toString()` for `UserRegisterDTO` class, so messages from above can work properly
```
@Override
public String toString() {
    return "{" +
        "username='" + username + '\'' +
        ", email='" + email + '\'' +
        ", firstName='" + firstName + '\'' +
        ", lastName='" + lastName + '\'' +
        "}";
}
```